### PR TITLE
connect.default on long avro field

### PR DESCRIPTION
**Pull request by drnushooz**
Fixes [1740](https://github.com/Apicurio/apicurio-registry/issues/1740)
Original fix https://github.com/confluentinc/schema-registry/issues/1376.

The issue happens with Avro 1.9+ where if a field of type `long` has `connect.default` property, e.g.
```
"type": "record"
"name": "sample"
"namespace": "io.apicurio",
"fields": [
  {
    "name": "prop",
    "type": {
      "type": "long",
      "connect.default": 42,
      "connect.version": 1
    }
  }
]
```

then `AvroData.toConnectSchema` throws
```
org.apache.kafka.connect.errors.DataException: Invalid type for INT64: class java.lang.Integer
	at io.apicurio.registry.utils.converter.avro.AvroData.toConnectData(AvroData.java:1498)
	at io.apicurio.registry.utils.converter.avro.AvroData.toConnectData(AvroData.java:1230)
	at io.apicurio.registry.utils.converter.avro.AvroData.toConnectData(AvroData.java:1478)
	at io.apicurio.registry.utils.converter.avro.AvroData.toConnectData(AvroData.java:1230)
	at io.apicurio.registry.utils.converter.avro.AvroData.toConnectData(AvroData.java:1226)
	at io.apicurio.registry.utils.converter.avro.AvroData.toConnectData(AvroData.java:1205)
	at com.github.drnushooz.kafka.connect.sqs.source.SqsSourceTask.lambda$poll$0(SqsSourceTask.java:138)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at com.github.drnushooz.kafka.connect.sqs.source.SqsSourceTask.poll(SqsSourceTask.java:149)
	at org.apache.kafka.connect.runtime.WorkerSourceTask.poll(WorkerSourceTask.java:265)
	at org.apache.kafka.connect.runtime.WorkerSourceTask.execute(WorkerSourceTask.java:232)
	at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:177)
	at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:227)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
```
The issue is caused by the change in the call to get the value of `connect.default`. The underlying API was switched from `schema.getJsonProp`, which returned a IntNode, to `schema.getObjectProp`, which returns an int. Later code that uses this default value differentiates between `JsonNode`s and non-`JsonNode`s. Depending on the type the treatment of the node is as follows.
* JsonNodes - Have their values extracted using type coercion to ensure default value matches the schema type.
* Non-JsonNodes are treated literally.  As the default is small enough to fit into an int, but the type is a long, the code throws the DataException.

The fix is to convert the returned default value back into a JsonNode so that the type coercion can happen.

### CODE REVIEW

File: utils/converter/src/main/java/io/apicurio/registry/utils/converter/avro/AvroData.java [hunk 1]
```diff
@@ -1487,7 +1488,7 @@ private Object toConnectData(Schema schema, Object value, ToConnectContext toCon
                     throw new DataException("Unknown Connect schema type: " + schema.type());
             }
 
-            if (schema != null && schema.name() != null && doLogicalConversion) {
+            if (schema.name() != null && doLogicalConversion) {
```
Comment (carlesarnal) [inline utils/converter/src/main/java/io/apicurio/registry/utils/converter/avro/AvroData.java:1491]: Why removing the null check from the schema name?

**Review by carlesarnal — APPROVED** [review]
LGTM, just a small curiosity to be addressed.

File: utils/converter/src/main/java/io/apicurio/registry/utils/converter/avro/AvroData.java (hunk 1 above)
Comment (drnushooz) [reply utils/converter/src/main/java/io/apicurio/registry/utils/converter/avro/AvroData.java:1491]: Based on the IDE analysis, the condition always appears to be true. To reduce the delta, I will add the null check back.

**Comment by drnushooz** [comment]
Thanks for merging this.

**Pull request merged**